scipy/optimize/_qap.py

```diff
@@ -0,0 +1,389 @@
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2].
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square, positive
+        A square adjacency matrix
+
+    dist_matrix : 2d-array, square, positive
+        A square adjacency matrix
+
+    seed_cost : 1d-array, optional, (default = [])
+        An array where each entry is an index of a node in 'cost_matrix'
+        (shape (m , 1) where m <= number of nodes).
+
+    seeds_dist : 1d-array, optional, (default = [])
+        An array where each entry is an index of a node in 'dist_matrix'
+        The elements of 'seed_cost' and 'seed_dist' are vertices which
+        are known to be matched, that is, 'seed_cost[i]' is matched to
+        vertex 'seed_dist[i]'. Array shape (m , 1)
+        where m <= number of nodes
+
+    maximize : bool (default = False)
+        Gives users the option to solve the Graph Matching Problem (GMP)
+        rather than QAP. This is accomplished through trivial negation
+        of the objective function.
+
+
+    options : dict, optional
+        A dictionary of solver options. All methods accept the following
+        options:
+
+            n_init : int, positive (default = 1)
+               Number of random initializations of the starting
+               permutation matrix that the FAQ algorithm will undergo.
+               n_init automatically set to 1 if init_method = 'barycenter'
+            init_method : string (default = 'barycenter')
+               The initial position chosen:
+               "barycenter" : the non-informative "flat doubly stochastic
+               matrix,":math:'J=1*1^T /n' , i.e the barycenter of the
+               feasible region
+               "rand" : some random point near :math:'J, (J+K)/2', where K
+               is some random doubly stochastic matrix
+            max_iter : int, positive (default = 30)
+               Integer specifying the max number of Franke-Wolfe iterations.
+               FAQ typically converges with modest number of iterations.
+            shuffle_input : bool (default = True)
+               Gives users the option to shuffle the nodes of 'cost_matrix'
+               to avoid results from inputs that were already matched.
+            eps : float (default = 0.1)
+               A positive, threshold stopping criteria such that FW
+               continues to iterate while Frobenius norm of
+               :math:'(P_{i}-P_{i+1}) > eps'
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` consisting of the fields:
+
+            col_ind : 1-D array
+                An array of column indices corresponding to the optimal optimal
+                permutation (with the fixed seeds given) on the
+                nodes of B, to best minimize the objective function
+                :math:'f(P) = trace(A^T PBP^T )'.
```

Comment:
It would be nice to explicitly specify how `P` can be calculated from `col_ind`, and/or show the fancy indexing that is actually used to compute `PBP^T` if the permutation matrix is not formed explicitly.